libraries/chain/controller.cpp

```diff
@@ -4162,9 +4166,9 @@ struct controller_impl {
                   const auto& bsp = *ritr;
 
                   br = controller::block_report{};
-                  apply_block( br, bsp, bsp->is_valid() ? controller::block_status::validated
-                                                        : controller::block_status::complete, trx_lookup );
-                  if (!switch_fork && check_shutdown()) {
+                  bool applied = apply_block( br, bsp, bsp->is_valid() ? controller::block_status::validated
+                                                                       : controller::block_status::complete, trx_lookup );
+                  if (!switch_fork && (!applied || check_shutdown())) {
```

Comment:
if `!applied` and we go into the `if`, we would call `shutdown()` a second time (we already called it in `apply_block()`.  Would that be an issue?

Reply:
No, it can be called multiple times.